apps/backend/src/application/application.service.ts

```diff
@@ -0,0 +1,121 @@
+import { BadRequestException, ForbiddenException, Injectable, NotFoundException, UseGuards } from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import { ApiBearerAuth } from '@nestjs/swagger';
+import { Application, Prisma, Role, User } from '@prisma/client';
+import { PrismaService } from 'nestjs-prisma';
+import { ApplicationPeriodService } from 'src/application-period/application-period.service';
+import { Roles } from 'src/auth/decorators/Roles.decorator';
+import { RolesGuard } from 'src/auth/roles.guard';
+
+import { CreateApplicationDto } from './dto/create-application.dto';
+import { UpdateApplicationDto } from './dto/update-application.dto';
+
+@Injectable()
+export class ApplicationService {
+  constructor(
+    private readonly prisma: PrismaService,
+    private readonly applicationPeriodService: ApplicationPeriodService
+  ) {}
+
+  async create(createApplicationDto: CreateApplicationDto, user: User) {
+    try {
+      return await this.prisma.application.create({
+        data: {
+          user: {
+            connect: {
+              authSchId: user.authSchId,
+            },
+          },
+          applicationPeriod: {
+            connect: {
+              id: createApplicationDto.applicationPeriodId,
+            },
+          },
+        },
+      });
```

Comment:
I think we should check whether the applicationPeriod is still accepting new applications